walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -15,9 +15,69 @@
 package com.google.sps;
 
 import java.util.Collection;
+import java.util.Arrays;
+import java.util.ArrayList;
 
 public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    if (request.getAttendees().size() <= 0) {
+        return Arrays.asList(TimeRange.WHOLE_DAY);
+    }
+    if (request.getDuration() > TimeRange.WHOLE_DAY.duration()) {
+        return Arrays.asList();
+    }
+    // check every *duration* length block of the day
+    int minutes = 0;
+    int length = 0;
+    int duration = (int) request.getDuration();
+    int start = TimeRange.getTimeInMinutes(0, 0);
+    boolean available;
+    int time;
+    TimeRange availableMeetingTime = null;
+    ArrayList<TimeRange> availableMeetingTimes = new ArrayList<TimeRange>();
+    TimeRange potentialMeetingTime;
+    while (minutes < (24 * 60)) {
```

Comment:
I don't think this is algorithmically efficient to loop over every minute of the day to check if that time slot is available. What if the problem needed you find a time slot over many days/years instead of over just a day?

Reply:
Resolving this after our discussion in the daily sync up.